mimic/test/test_identity_objects.py

```diff
@@ -0,0 +1,302 @@
+            uuid=eeapi_template_id
+        )
+        alternate_eeapi_template = ExampleEndpointTemplate(
+            name=self.eeapi_name,
+            uuid=alternate_eeapi_template_id
+        )
+        eeapi = make_example_external_api(
+            name=self.eeapi_name,
+            set_enabled=True,
+            endpoint_templates=[
+                eeapi_template,
+                alternate_eeapi_template
+            ]
+        )
+        eeapi.enable_endpoint_for_tenant(
+            'some-tenant',
+            eeapi_template_id
+        )
+        eeapi.enable_endpoint_for_tenant(
+            'some-other-tenant',
+            alternate_eeapi_template_id
+        )
+
+        eeapi.remove_template(eeapi_template_id)
+
+
+class EndpointsForTenants(SynchronousTestCase):
+    """
+    Tests for functionality specific to tenants
+    """
+    def setUp(self):
+        self.eeapi_name = u"externalServiceName"
+
+    def test_invalid_template_endpoint_enable(self):
+        """
+        Validate enabling an invalid endpoint template
+        for a user raises ValueError
+        """
+        eeapi = make_example_external_api(
+            name=self.eeapi_name,
+            set_enabled=True
+        )
+
+        with self.assertRaises(ValueError):
+            eeapi.enable_endpoint_for_tenant(
+                'some_tenant',
+                'some-invalid-template-id'
+            )
+
+    def test_invalid_template_endpoint_disable(self):
+        """
+        Validate disabling an invalid endpoint template
+        for a user raises ValueError
+        """
+        eeapi = make_example_external_api(
+            name=self.eeapi_name,
+            set_enabled=True
+        )
+
+        with self.assertRaises(ValueError):
+            eeapi.disable_endpoint_for_tenant(
+                'some_tenant',
+                'some-invalid-template-id'
+            )
+
+    def test_disable_endpoint_template_for_tenant(self):
+        new_url = "https://api.new_region.example.com:9090"
+        new_region = "NEW_REGION"
+        new_eeapi_template_id = u"uuid-alternate-endpoint-template"
+        new_eeapi_template = ExampleEndpointTemplate(
+            name=self.eeapi_name,
+            uuid=new_eeapi_template_id,
+            region=new_region,
+            url=new_url
+        )
+        eeapi = make_example_external_api(
+            name=self.eeapi_name,
+            set_enabled=True,
+            endpoint_templates=[new_eeapi_template]
+        )
+        eeapi.enable_endpoint_for_tenant(
+            'some-tenant',
+            new_eeapi_template_id
+        )
+        eeapi.disable_endpoint_for_tenant(
+            'some-tenant',
+            new_eeapi_template_id
+        )
+
+
+class EndpointTemplateOperations(SynchronousTestCase):
+    """
+    Tests for creating a :class:`MimicCore` object with apis
+    """
+    def setUp(self):
+        self.eeapi_name = u"externalServiceName"
+
+    def test_uri_for_service_with_invalid_region(self):
```

Comment:
Could you explain to me what this test covers? (I ask because it isn't immediately obvious to me why it isn't in mimic/test/test_core.py, but I could definitely be missing something :-) )